Microsoft.Toolkit.Uwp/Helpers/DispatcherHelper.cs

```diff
@@ -258,12 +260,27 @@ public static Task<T> AwaitableRunAsync<T>(this CoreDispatcher dispatcher, Func<
         /// <returns>Awaitable Task</returns>
         public static Task AwaitableRunAsync(this CoreDispatcher dispatcher, Action function, CoreDispatcherPriority priority = CoreDispatcherPriority.Normal)
         {
-            return dispatcher.AwaitableRunAsync<object>(
-                () =>
+            if (function == null)
+            {
+                throw new ArgumentNullException("function can't be null!");
+            }
+
+            var taskCompletionSource = new TaskCompletionSource<object>();
```

Comment:
Same here. Having to deal with the same code that `Func<T>` has already had. More relatively-same code to maintain imo.